include/examples/quadruped_3dof_limb.h

```diff
@@ -0,0 +1,323 @@
+/**
+ * @file quadruped_3dof_limb.h
+ * @author Kodlab - J. Diego Caporale (jdcap@seas.upenn.edu)
+ * @brief A LimbBase derived class example of the common quadrupedal serial limb
+ * configuration with a point toe
+ * @details 3-DoF limb implementation demonstrating the LimbBase class
+ * @date 2022-09-02
+ *
+ * @copyright (c) Copyright 2022 The Trustees of the University of Pennsylvania
+ *
+ */
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include "kodlab_mjbots_sdk/limb_base.h"
+
+namespace quad_example_helpers{
+    
+/**
+ * @brief Creates a 3x3 skew symmetric matrix from a 3-vector
+ *
+ * @param w 3-vector
+ * @return 3x3 skew-symmetric matrix
+ */
+Eigen::Matrix3f Skew(const Eigen::Vector3f &w)
+{
+    Eigen::Matrix3f S;
+    S <<    0, -w(2),  w(1),
+         w(2),     0, -w(0),
+        -w(1),  w(0),     0;
+    return S;
+}
+
+/**
+ * @brief Get the sign of a flag for use in mathematic equations
+ *
+ * @param flag boolean flag
+ * @return 1 if `flag` evaluates to true, -1 otherwise
+ */
+static int GetSign(bool flag) { return (flag ? 1 : -1); }
+
+/**
+     * @brief Solving the equation a0 Sin(th) + a1 Cos(th) = a2
+     *
+     * @param a0 Sin coefficient
+     * @param a1 Cos coefficient
+     * @param a2 Right side coefficient
+     * @return float returns nan if no solution exists
+*/
+float SolveSinCosSum(float a0, float a1, float a2)
+{
+    // a0 Sin(x) + a1 Cos(x) = a2 Solution for x
+    float sqDiff = a0 * a0 + a1 * a1 - a2 * a2;
+    if (sqDiff < 0)
+    { // Check that solution is feasible
+        std::cout << "ERROR NO SOLUTION SinCosSum" << std::endl;
+        return 0.0 / 0.0; // NaN
+    }
+
+    // Mathematica Code:
+    //  ArcTan[a1*a2 + Sqrt[a0^2 + a1^2 - a2^2]*Abs[a0], 
+    //         a0*a2 - (a1*Sqrt[a0^2 + a1^2 - a2^2])/Sign[a0]]
+    float diff = sqrt(sqDiff);
+    return atan2(a0 * a2 - a1 * diff / std::copysignf(1.0, a0), 
+                 a1 * a2 + diff * std::abs(a0));
+}
+
+/**
+ * @brief Wraps an angle -Pi to Pi
+ * @param angle
+ */
+float angle_mod(float angle){
+    return fmod(angle+M_PI, 2*M_PI) - M_PI;
+}
+
+/**
+ * @brief Quick Float Nan
+ * 
+ */
+float qnanf(){
+    return float(0.0/0.0);
+}
+} //namespace example_helpers
+
+/**
+ * @brief A 3-DoF limb based on the common serial linkage in 12DoF quadrupeds
+ * @details Based on the limbs of popular quadrupeds
+ * @warning Needs to be tested, use at your own risk!!!!
+ * @todo    Needs to be tested ^^^
+ * @note    Needs to be tested ^^^
+ */
+class QuadrupedLimb : public kodlab::LimbBase<Eigen::Vector3f> {
```

Comment:
How will templating this to 3f play with the robot base using xf?

Reply:
I think it should work based on my understanding, but I don't know. I am rethinking having the Limbs vector at all and would love to figure that out, perhaps separate from this PR? i.e. remove it all together for now.

Reply:
I think thats fine.